lib/smooth/application/io/spi/BME280SPI.cpp

```diff
@@ -94,7 +94,12 @@ namespace smooth::application::sensor
 
     uint8_t BME280SPI::read_id()
     {
-        alignas(4) std::array<uint8_t, 4> rxdata;
+        std::array<uint8_t, 4> rxdata;
+
+        // The spi rx buffer need to be length of multiples of 32 bits to avoid heap corruption.
+        // see esp-idf/components/driver/spi_master.c line 865
+        //static_assert(&rxdata % 4 == 0, "Error:c rxdata not 32 bit aligned ");
+
```

Comment:
You can do it like this:
```
using BuffType = std::array<uint8_t, 4>;
BuffType foo{};
static_assert(sizeof(BuffType::value_type) * foo.size() % 4 == 0, "Buffer must have a length of multiple of 32 bits, i.e. 4 bytes");
```

You can then place the static assert at a single place where BuffType is used to declare a member. Just make sure to name the type better than my example :)

Reply:
Went with different appoach